### PR TITLE
[AFF] Custom AJSON filter for INTF serialization

### PR DESCRIPTION
**Pull request by albertmink**
Ref #5658
Aims to emulate behavior described in [AFF](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#default-values), `$default` and other annotations

### CODE REVIEW

**Comment by albertmink** [comment]
https://github.com/sbcgua/ajson/pull/111 is required to set values without changing the path (camel case).
Actually, `originalLanguage` becomes `originallanguage` after setting a custom value via `set` method.

**Comment by sbcgua** [comment]
https://github.com/sbcgua/ajson/pull/111 merged.

P.S. potentially the behavior may change in future so that conversion is applied at parsing time. Then it will create a bug as the access path will be `original_language` not `originalLanguage`. Anyway for the moment the ajson is unblocked

**Comment by larshp** [comment]
https://github.com/abapGit/abapGit/pull/5682

File: src/objects/aff/zcl_abapgit_json_handler.clas.abap [hunk 1]
```diff
@@ -75,15 +111,81 @@ CLASS zcl_abapgit_json_handler IMPLEMENTATION.
...
 
     lv_json = lo_ajson_filtered->stringify( 2 ).
 
     rv_result = zcl_abapgit_convert=>string_to_xstring_utf8( lv_json ).
+  ENDMETHOD.
+
+  METHOD set_original_language.
+    DATA:
+      lv_iso_language      TYPE i18_a_langiso2,
+      lv_original_language TYPE sylangu.
```
Comment (larshp) [inline src/objects/aff/zcl_abapgit_json_handler.clas.abap:134]:
```suggestion
      lv_original_language TYPE sy-langu.
```

**Comment by albertmink** [comment]
@larshp can you update the ASJON inside abapGit? The ordering is as expected and has been merged, see https://github.com/sbcgua/ajson/pull/115

**Comment by larshp** [comment]
done

**Comment by albertmink** [comment]
Let me add a short video
https://user-images.githubusercontent.com/38354196/180751465-a6244542-842e-4de6-8b1e-716cfbe62087.mp4

**Review by larshp — APPROVED** [review]

**Comment by larshp** [comment]
Few notes:
* please add unit tests
* we need to generalize/move some of the json parts to somewhere generic

File: src/objects/zcl_abapgit_object_intf.clas.locals_imp.abap [hunk 2]
```diff
@@ -461,37 +461,94 @@ CLASS lcl_aff_type_mapping IMPLEMENTATION.
 
 ENDCLASS.
 
+
 CLASS lcl_aff_serialize_metadata DEFINITION.
   PUBLIC SECTION.
 
     CLASS-METHODS serialize
       IMPORTING is_intf          TYPE data
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_intf.clas.locals_imp.abap:469]: can this IS_INTF get the correct type?

Comment (albertmink) [reply src/objects/zcl_abapgit_object_intf.clas.locals_imp.abap:469]: I guess so. At some point in time, there was an underlying interface, that had TYPE data

**Comment by albertmink** [comment]
> * we need to generalize/move some of the json parts to somewhere generic

Any suggestions, starting points?

**Comment by larshp** [comment]
lets get this PR merged first, then we can take a look

I'll merge it tomorrow morning(EU time) if there are no additional comments

**Pull request merged**